hydrogym/core.py

```diff
@@ -355,19 +422,43 @@ def step(
     """Advance the state of the environment.  See gym.Env documentation
 
         Args:
-            action (Iterable[ArrayLike], optional): Control inputs. Defaults to None.
+            action (Iterable[ActType], optional): Control inputs. Defaults to None.
 
         Returns:
-            Tuple[ArrayLike, float, bool, dict]: obs, reward, done, info
+            Tuple[ObsType, float, bool, dict]: obs, reward, done, info
         """
-    self.solver.step(self.iter, control=action)
-    self.iter += 1
-    t = self.iter * self.solver.dt
+    action = action * self.flow.CONTROL_SCALING
+
+    if self.num_sim_substeps_per_actuation is not None and self.num_sim_substeps_per_actuation > 1:
+      self.action = action
+      self.flow = self.solver.solve_multistep(
+          num_substeps=self.num_sim_substeps_per_actuation,
+          callbacks=[self.rewardLogCallback],
+          controller=self.constant_action_controller,
+          start_iteration_value=self.iter)
+      if self.reward_aggreation_rule == "mean":
+        averaged_objective_values = np.mean(self.flow.reward_array, axis=0)
+      elif self.reward_aggreation_rule == "sum":
+        averaged_objective_values = np.sum(self.flow.reward_array, axis=0)
+      elif self.reward_aggreation_rule == "median":
+        averaged_objective_values = np.median(self.flow.reward_array, axis=0)
+      else:
+        raise NotImplementedError(
+            f"The {self.reward_aggreation_rule} function is not implemented yet."
+        )
+
+      self.iter += self.num_sim_substeps_per_actuation
+      self.t += self.dt * self.num_sim_substeps_per_actuation
+      reward = self.get_reward(averaged_objective_values)
+    else:
+      self.solver.step(self.iter, control=action)
+      self.iter += 1
+      self.t += self.dt
+      reward = self.get_reward()
+
```

Comment:
Is the `else` statement above redundant since `num_sim_substeps_per_actuation == 1` should be equivalent? (Not critical, and if things are in "production" it may be dangerous to modify the code too much...)